src/services/user/model/user.schema.js

```diff
@@ -107,6 +109,23 @@ if (Configuration.get('FEATURE_TSP_ENABLED') === true) {
 	userSchema.index({ 'sourceOptions.$**': 1 });
 }
 
+// This 'pre-save' method slices the firstName, lastName and email
+// To allow searching the users
+userSchema.pre('save', function() {
+	const arr = [];
+	const firstName = this.firstName.replace(/\s/g,'');
```

Comment:
there is a trim command for it, no need for regex. What you currently do is removing all whitespace so if you have a first name like "hans peter" you will have "hanspeter"